### ISSUE TITLE
Force serialization to not use `Codable`?

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/Defaults#79]
It would be useful to have a way to force the serialization to not use the `Codable` implementation.

Let's say I have an enum:

```swift
enum Foo: String, CaseIterable, Codable, Defaults.Serializable {
	case bar
}
```

It needs to stay `Codable` as I need to be able to use it for some other purposes too, where `Codable` is needed. This also applies to types you don't control that are already `Codable`.

However, I would like to store it using the native enum handling instead of using `Codable` (so it's not wrapped in double-quotes). It would be nice if we had a way to force that.

This would also be a nice safe-guard. Imagine you have a type that is `Defaults.Serializable`, but not `Codable`, and later on you add `Codable` conformance to it, then the stored representation would change and you could accidentally lose the data.

@@hank121314 Thoughts?

**Comment by hank121314** [comment]
This might be the same feature as you mentioned in #80?
> It would be nice if a user could somehow choose whether to use the Codable or NSSecureCoding for the serialization.

**Comment by sindresorhus** [comment]
> This might be the same feature as you mentioned in #80?

Yes, just wanted to point out another scenario where it would have been useful.

**Comment by hank121314** [comment]
Maybe we can try to add a `usingCodable` property in `Defaults.Key`?

For example.
```
let key = Defaults.Key<Foo>(default: Foo.bar, usingCodable: false)
```

I am working on this will create a PR later!

**Comment by sindresorhus** [comment]
I was thinking it would be in the type system:

```swift
enum Foo: String, CaseIterable, Codable, Defaults.Serializable {
	case bar
}

// And in my code:
extension Foo: Defaults.NoCodable {}

extension Defaults.Keys {
	static let foo = Key<Foo>("foo", default: .bar)
}
```

The benefit of your idea is that it applies per key and not per type, but mine is definitely simpler, and I cannot think of many situations where you want it per key and not just per type.

And if you do need it per key, could something like this work?

```swift
enum Foo: String, CaseIterable, Codable, Defaults.Serializable {
	case bar
}

// And in my code:
typealias FooNoCodable = Foo & Defaults.NoCodable

extension Defaults.Keys {
	static let foo = Key<FooNoCodable>("foo", default: .bar)
}
```

If `Defaults.NoCodable` has no requirements, it could be used as a "marker protocol", but I'm honestly not sure if `Foo` would still be assignable to it.

I'm not saying mine is better, but it would be good to discuss this a bit, to explore our options.

**Comment by hank121314** [comment]
> The benefit of your idea is that it applies per key and not per type, but mine is definitely simpler, and I cannot think of many situations where you want it per key and not just per type.

That's true! Per key might be useless.

For per type, I have some thoughts.
We can public the `Defaults` internal bridge(ex. `RawRepresentableBridge` or `NSSecureCodingBridge`) to users.
Users can choose the bridge with static let `bridge` in `Defaults.Serializable`.

If we want a `Foo` enum type that will always use `RawRepresentableBridge`, we can do it like this.

```
enum Foo: String, CaseIterable, Codable, Defaults.Serializable {
	case bar
	public static let bridge = Defualts.RawRepresentableBridge<Self>()
}
```

**Comment by sindresorhus** [comment]
> If we want a Foo enum type that will always use RawRepresentableBridge, we can do it like this.

Yeah, that's true. I would still like to simplify the common case though, for example, choosing between `RawRepresentable` and `Codable`. Could we do a `enum Foo: String, CaseIterable, Codable, Defaults.RawRepresentableSerializable` type that makes the user not have to do `public static let bridge = Defualts.RawRepresentableBridge<Self>()`?

Alternatively: `enum Foo: String, CaseIterable, Codable, Defaults.Serializable, Defaults.PreferRawRepresentable`.
